src/fragment-components/a-radiobutton.tsx

```diff
@@ -0,0 +1,155 @@
+		...fragment,
+		data: updatedState(
+			fragment.data,
+			{
+				type: 'insert',
+				component: {
+					type: 'radioButton',
+					value: `${componentObj.id}`,
+					codeContext: {
+						formItemName: componentObj.codeContext?.formItemName
+					},
+					labelText: 'New Option',
+					disabled: componentObj.disabled
+				}
+			},
+			parentComponent.id,
+			parentComponent.items.indexOf(componentObj) + offset
+		)
+	});
+	componentObj.value = `${componentObj.id}`;
+	return (<>
+		<Adder
+			active={selected}
+			addButtonsCss={addButtonCss}
+			key={componentObj.id}
+			bottomAction={() => addRadio(1)}>
+			<AComponent
+				selected={selected}
+				headingCss={css`width: fit-content; min-width: 9rem;`}
+				componentObj={componentObj}
+				{...rest}> 
+					<RadioButton
+						id={componentObj.id}
+						name={componentObj.codeContext?.formItemName}
+						labelText={componentObj.labelText}
+						defaultChecked={componentObj.defaultChecked}
+						checked={componentObj.defaultChecked}
+						value= {componentObj.value}
+						disabled= {componentObj.disabled}/>
+			</AComponent>
+		</Adder>
+	
+	</>);
+};
+
+export const componentInfo: ComponentInfo = {
+	component: ARadioButton,
+	styleUI: ARadioButtonStyleUI,
+	codeUI: ARadioButtonCodeUI,
+	render: ({ componentObj, select, remove, selected }) => <ARadioButton
+	componentObj={componentObj}
+	select={select}
+	remove={remove}
+	selected={selected}>
+		{componentObj.labelText}
+	</ARadioButton>,
+	keywords: ['radio','button'],
+	name: 'Radio button',
+	defaultComponentObj: {
+		type: 'radioButton'
+	},
+	image: image,
+	hideFromElementsPane: true,
+	codeExport: {
+		angular: {
+			inputs: ({json}) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Label = "${json.labelText}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Disabled = ${json.disabled};
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Id = "${json.codeContext?.name}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Value = "${json.value}";
+								@Input() ${nameStringToVariableString(json.codeContext?.name)}Checked = ${json.defaultChecked};`,
+            outputs: ({json}) => ``,
+            imports: ['RadioModule'],
+			code: ({json }) => {
+				return `<ibm-radio
+					[id]="${nameStringToVariableString(json.codeContext?.name)}Id"
+					[value]="${nameStringToVariableString(json.codeContext?.name)}Value"
+					[checked]="${nameStringToVariableString(json.codeContext?.name)}Checked"
+					[disabled]="${nameStringToVariableString(json.codeContext?.name)}Disabled"
+					${angularClassNamesFromComponentObj(json)}>
+					{{${nameStringToVariableString(json.codeContext?.name)}Label}}
+				</ibm-radio>`;
+			}
+		},
+		react: {
+			imports: ['RadioButton'],
+			code: ({ json }) => {
+				return `<RadioButton
+					id="${json.codeContext?.name}"
+					value="${json.value}"
+					checked="${json.defaultChecked}"
```

Comment:
Should be curly braces since we are assigning a boolean to prop - Also may want to add a condition where if it is false, not append the prop to reduce unnecessary props being passed also increases code length 

```suggestion
					checked={${json.defaultChecked}}
```

Reply:
the checked value should always be passed to determine the state of the radiobutton, otherwise it'll be null